src/main/jbake/content/blog/2017/02/03/grafana-events.adoc

```diff
@@ -0,0 +1,114 @@
+= Display custom events in Grafana
+Joel Takvorian
+2017-02-03
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, grafana, events
+
+The latest release of the link:https://github.com/hawkular/hawkular-grafana-datasource/releases/tag/v1.0.7[Hawkular Grafana plugin] (1.0.7) introduced a new feature, implementing link:http://docs.grafana.org/reference/annotations/[Grafana Annotations].
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-02-03-annotations-tooltip-2.png[Logged error]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-02-03-annotations-tooltip-2.png[Logged error]
+endif::[]
+_A logged error, associated with a tag_
+
+In Grafana, annotations allow to display vertical bars on charts to illustrate an event in time, with a dedicated tooltip containing a custom text and eventually tags. All annotations are shared across graphs in a given dashboard.
+
+In hawkular-metrics, events can be stored in link:http://www.hawkular.org/hawkular-metrics/docs/user-guide/#_metric_types[String metrics]. You can send string datapoints, just like you would send gauge or counter datapoints on other metrics. Simply note that these strings are currently limited to 2 KB weight.
+
+As a concrete example, we will reuse the sample application that I described link:http://www.hawkular.org/blog/2017/01/16/hawkular-metrics-with-dropwizard.html[in a previous post], which creates metrics through dropwizard with the pretext of a small benchmarking scenario.
+It will be modified to generate some events: scenario steps and error logs.
+
+The scenario consisted in running two cache implementations (guava and ehcache), backed by a fake database, on which _get_ accesses are being monitored. We'll introduce random errors in DB access, that will be caught and logged to Hawkular.
+
```

Comment:
You could add a === headline here to say "Changes in the scenario
and below "Set up in Grafana"